### PR TITLE
Add conditional return in `handleRemoveView`

### PR DESCRIPTION
**Pull request by bartlomiejbloniarz**
## Summary:

I was recently working on an [issue](https://github.com/software-mansion/react-native-reanimated/issues/5715) in Reanimated where z-index of some views was broken after a Layout Animation was used. The issue was that in some cases we were calling the `removeView` function on a already removed view. On plain Android this wouldn't be an issue, since the `removeView` function ignores such calls. Unfortunately, the `ReactViewGroup.java` implementation maintains a counter of views with user defined z-index. This counter is decremented whenever a call to `removeView` is made, even if the view is not a child of this `ViewGroup`. This PR adds an additional check in the `handleRemoveView` function to unify the `removeView` behavior between Android and react-native.

## Changelog:
[ANDROID] [CHANGED] - Changed the `handleRemoveView` function in `ReactViewGroup.java` to ignore calls for `Views` that are not children of this `ViewGroup`

## Test Plan:

I tested if the `rn-tester` app behaves correctly after those changes.

### CODE REVIEW

**Comment by javache** [comment]
Does this issue only affect the old architecture or does it repro on the new arch too? (zonder is handled differently there)

**Comment by bartlomiejbloniarz** [comment]
I tested it only on the old architecture, but I think this issue should only be present there.

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java [hunk 1]
```diff
@@ -518,6 +518,10 @@ private void handleAddView(View view) {
   private void handleRemoveView(View view) {
     UiThreadUtil.assertOnUiThread();
 
+    if (indexOfChild(view) == -1){
+      return;
+    }
+
     if (!customDrawOrderDisabled()) {
```
Comment (javache) [inline packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java:525]: Can you move this `indexOfChild` check inside this if-branch? That way we avoid the check in the new architecture.

Comment (bartlomiejbloniarz) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java:521]: Thanks, that's a good idea

**Comment by bartlomiejbloniarz** [comment]
Hi @fabriziocucci. There might have been a small miscommunication on my part. I tested the code from the snack only within our ReanimatedExampleApp. Then I applied those changes in the forked `react-native` repo and tested if the `rn-tester` app still behaves correctly (i.e. if views are removed normally). 

After some testing I think the issue that you are seeing is caused by the fact that our compatibility with bridgeless is still work in progress, so using reanmiated in the `rn-tester` app will produce those errors.

**Pull request closed without merging**